platform/disk/ephemeral_device_partitioner.go

```diff
@@ -0,0 +1,192 @@
+	deviceUtil        Util
+	logger            boshlog.Logger
+
+	logTag      string
+	cmdRunner   boshsys.CmdRunner
+	fs          boshsys.FileSystem
+	timeService clock.Clock
+}
+
+type Settings struct {
+	AgentID string `json:"agent_id"`
+}
+
+func NewEphemeralDevicePartitioner(
+	partedPartitioner Partitioner,
+	deviceUtil Util,
+	logger boshlog.Logger,
+	cmdRunner boshsys.CmdRunner,
+	fs boshsys.FileSystem,
+	timeService clock.Clock,
+) *EphemeralDevicePartitioner {
+	return &EphemeralDevicePartitioner{
+		partedPartitioner: partedPartitioner,
+		deviceUtil:        deviceUtil,
+		logger:            logger,
+		logTag:            "EphemeralDevicePartitioner",
+		cmdRunner:         cmdRunner,
+		fs:                fs,
+		timeService:       timeService,
+	}
+}
+
+func (p *EphemeralDevicePartitioner) Partition(devicePath string, partitions []Partition) error {
+	existingPartitions, deviceFullSizeInBytes, err := p.partedPartitioner.GetPartitions(devicePath)
+	if err != nil {
+		return bosherr.WrapErrorf(err, "Getting existing partitions of `%s'", devicePath)
+	}
+
+	if p.matchPartitionNames(existingPartitions, partitions, deviceFullSizeInBytes) {
+		p.logger.Info(p.logTag, "%s already partitioned as expected, skipping", devicePath)
+		return nil
+	}
+
+	err = p.removePartitions(existingPartitions, devicePath)
+	if err != nil {
+		return bosherr.WrapErrorf(err, "Removing existing partitions of `%s'", devicePath)
+	}
+
+	err = p.ensureGPTPartition(devicePath)
+	if err != nil {
+		return bosherr.WrapErrorf(err, "Ensuring GPT table of `%s'", devicePath)
+	}
+
+	return p.partedPartitioner.Partition(devicePath, partitions)
+}
+
+func (p *EphemeralDevicePartitioner) GetDeviceSizeInBytes(devicePath string) (uint64, error) {
+	return p.partedPartitioner.GetDeviceSizeInBytes(devicePath)
+}
+
+func (p *EphemeralDevicePartitioner) GetPartitions(devicePath string) (partitions []ExistingPartition, deviceFullSizeInBytes uint64, err error) {
+	return p.partedPartitioner.GetPartitions(devicePath)
+}
+
+func (p *EphemeralDevicePartitioner) matchPartitionNames(existingPartitions []ExistingPartition, desiredPartitions []Partition, deviceSizeInBytes uint64) bool {
+	if len(existingPartitions) < len(desiredPartitions) {
+		return false
+	}
+
+	for index, partition := range desiredPartitions {
+		existingPartition := existingPartitions[index]
+
+		if !strings.HasPrefix(existingPartition.Name, partition.NamePrefix) {
+			return false
+		}
+
+	}
+
+	return true
+}
+
+func (p EphemeralDevicePartitioner) removePartitions(partitions []ExistingPartition, devicePath string) error {
+	partitionPaths, err := p.getPartitionPaths(devicePath)
+	if err != nil {
+		return bosherr.WrapErrorf(err, "Getting partition paths of disk `%s'", devicePath)
+	}
+
+	p.logger.Debug(p.logTag, "Erasing old partition paths")
+	for _, partitionPath := range partitionPaths {
+		partitionRetryable := boshretry.NewRetryable(func() (bool, error) {
+			_, _, _, err := p.cmdRunner.RunCommand(
+				"wipefs",
```

Comment:
Yes, it is format after partition. But if reload VM, the agent needs to wipe fs otherwise we could not remove old partitions and then repartition this device.

I think `wipefs` is part of util-linux(-ng) and already installed in each stemcell(centos, trusty and xenial).